IdentityCore/src/webview/embeddedWebview/ui/ios/MSIDNTLMUIPrompt.m

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDNTLMUIPrompt.h"
+#import "MSIDAppExtensionUtil.h"
+#import "UIApplication+MSIDExtensions.h"
+
+@implementation MSIDNTLMUIPrompt
+
+__weak static UIAlertController *_presentedPrompt = nil;
+
++ (void)dismissPrompt
+{
+    dispatch_async(dispatch_get_main_queue(), ^{
+        
+        if (_presentedPrompt.presentingViewController)
+        {
+            [_presentedPrompt.presentingViewController dismissViewControllerAnimated:YES completion:nil];
+        }
+        
+        _presentedPrompt = nil;
+    });
+}
+
++ (void)presentPrompt:(void (^)(NSString *username, NSString *password, BOOL cancel))block
+{
+    
+    if ([MSIDAppExtensionUtil isExecutingInAppExtension])
+    {
+        block(nil, nil, YES);
+        return;
+    }
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
do we need this big of a dispatch block?

Reply:
Most of them are from UIKit. It is recommended to use main thread when accessing UIKit classes :)
`Use UIKit classes only from your app’s main thread or main dispatch queue, unless otherwise indicated. `(https://developer.apple.com/documentation/uikit?language=objc)